nx/lib/nx/binary_backend/matrix.ex

```diff
@@ -3,40 +3,140 @@ defmodule Nx.BinaryBackend.Matrix do
   @default_eps 1.0e-10
   import Nx.Shared
 
-  def ts(a_data, a_type, b_data, b_type, {rows, rows} = shape, output_type, _opts) do
-    a_matrix = binary_to_matrix(a_data, a_type, shape)
-    b_matrix = binary_to_matrix(b_data, b_type, shape)
+  def ts(a_data, a_type, b_data, b_type, shape, output_type, opts) do
```

Comment:
@josevalim I ended up implementing this "properly" because with a few row-swapping and transposing tricks the base `ts` code could solve all of the new options.

It should be noted that I haven't generated any examples using Jax, so maybe some results should be checked?

Reply:
Also also, this PR doesn't add the new opts to EXLA. We need to add them in a separate pull request. Most are straightforward, but the `transpose` option didn't seem to be a plain boolean.